int32 data
---
bool suc